### DISCUSSION TITLE
NMEA0183, USB connection

### DISCUSSION THREAD

**Discussion by Techstyleuk** [peterantypas/maiana#95] [Q&A]
I just ordered this device (Thanks Peter) and I am thinking about how I will set it up.  I have ordered the NMEA0183 version and plan to connect the USB to the Openplotter running RPI and have the NMEA0183 connected to a VHF radio.  I expect the VHF to read the info from Maiana and get AIS and GPS info from it.  The messages the VHF will listen to are listed below under NMEA0183-HS input.  In addition the VHF will output the DSC and DSE sentences - will Maiana pass these through to the USB or do I need to connect in a RS422/USB from the VHF to the Pi?

![image](https://user-images.githubusercontent.com/79181587/154195830-fd69812a-b138-4f93-962f-0e07d71ebd70.png)

Thanks for the Great work, by the way!

**Comment by peterantypas** [comment]
MAIANA is a talker only, you’ll need to feed the Pi via some other adapter.

**Comment by muebau** [comment]
@Techstyleuk just out of curiosity, what do you with the DSC and DSE sentences.
I was in hope to find some cool GUI to see all the recent DSC calls in OpenCPN. I found nothing.

I thought it would be nice to have some kind of "recent radio" thing eg. in the browser.

If combined with some SDR (2 rtlsdr should be enough) there could be even a recording of all channels at the same time and have them on some multichannel timeline. If you would click on an item you hear the radio transmission and you see the ATIS (Europe) etc.

**Reply by kagouraki** [reply]
I dont think there is a plugin of any kind to do what you want... 
This would be VERY useful though. If you find a solution i would be interested!!

**Reply by muebau** [reply]
Well I think I would be possible to build a simple plugin for SignalK and have a list of DSC info as it comes by.

The SDR idea needs some GNU radio pieces to separate the channels etc. like done in WebSDR apps.

**Reply by Techstyleuk** [reply]
It looks like the message is only sent when someone is transmitting, and it is clear if they have to be hailing you directly or just general transmissions.  Either way it would be nice if the chart plotter could display who was calling and nice if there 'boat' would appear (like an AIS triangle) and flash to say they were transmitting.  They may not have AIS and this would be another way to 'see' traffic

**Reply by muebau** [reply]
This is a very good idea. If something like Navionics or iNavX is used it would be shown if the DSC information is translated to an AIS sentence to be shown on the map.

It looks like there is no usage of the DSC sentences in OCPN/Navionics/iNavX/etc. for some reason.
Most radios have no way to do simple thinks like to change the channel with some NMEA0183 sentence. If there would be some way to do it OpenCPN/ etc. could be used to utilize the DSC/AIS information to call another vessel or hail them etc.

**Reply by kagouraki** [reply]
Automated usage of the vhf is risky. 
But the option to "record" dsc calls is an interesting one. 

Peter, do you think maiana would be capable of listening to two changable channels except the two AIS ones ? If yes, then through the plugin you could set the two desired channels and some how maiana could forward the transmission to the plugin and the plugin saves it in the raspberry so you can replay it. What do you think ?

**Reply by peterantypas** [reply]
MAIANA can only decode (G)FSK modulated digital signals.

**Reply by kagouraki** [reply]
If it is a hardware problem ok. But if it is a software problem then...

**Reply by peterantypas** [reply]
Yes, this is a hardware limitation. I'm not aware of any inexpensive and low-power silicon that can decode DSC. It will have to be some kind of SDR with a lot of post-processing.

**Reply by kagouraki** [reply]
Dead end then...

**Reply by Techstyleuk** [reply]
@muebau - it seems like some of this functionality is available.  according to this page: 
https://opencpn.org/wiki/dokuwiki/doku.php?id=opencpn:opencpn_user_manual:toolbar_buttons:ais:dsc_buddy_via_ais
DSC messages are treated similar to AIS and will appear in the AIS target list and will be highlighted on the screen.

**Reply by muebau** [reply]
Oh wow you are right. This is very cool.

**Reply by muebau** [reply]
> Automated usage of the vhf is risky. But the option to "record" dsc calls is an interesting one.

Well some of the commercial plotter support the "automated usage" as I mean it. One can click on the screen (AIS etc.) and the VHF sends a DSC call request. This is not very risky.

The "record" is a nice thing and should be not to complicated to implement if 2 RTLSDR would be used to cover the complete VHF band.
There are ready to use GNU radio blocks to cut a band in several channels, make a power analysis on every channel and output just the currently "active" channels in one step.
From there every call on every channel could be stored in a separate file. I have seen GNU radio blocks for most of this. The question is if a Raspberry Pi has enough power to lift this.
